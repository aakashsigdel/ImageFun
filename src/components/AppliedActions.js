import React from 'react';

import './Actions.scss';

const AppliedActions = props =>
  <div className="applied-actions-container">
    {props.appliedActions.map(action =>
      <div key={action.id}
        className="action"
        onClick={() => props.popAction(action)}
      >
        {action.name}
      </div>
    )}
  </div>;

export default AppliedActions;
