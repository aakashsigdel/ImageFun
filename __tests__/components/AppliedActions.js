import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';

import AppliedActions from '../../src/components/AppliedActions';

import renderer from 'react-test-renderer';

describe('<AppliedActions /> component', () => {
  it('should render <AppliedActions /> correctly', () => {
    const tree = renderer.create(<AppliedActions appliedActions={[]} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should call popActions onClick', () => {
    const popAction = sinon.spy();
    const wrapper = shallow(
      <AppliedActions
        appliedActions={[{id: 1, action: 'roatate', name: 'Rotate'}]}
        popAction={popAction}
      />);
    wrapper.find('.action').simulate('click');

    expect(popAction.calledOnce).toEqual(true);
  })
});


