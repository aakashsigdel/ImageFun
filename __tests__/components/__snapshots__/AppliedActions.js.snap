exports[`<AppliedActions /> component should render <AppliedActions /> correctly 1`] = `
<div
  className="applied-actions-container">
  <div
    className="menu-title">
    Applied Actions
  </div>
  <ul />
</div>
`;
